MergerService/src/run.cs

```diff
@@ -51,153 +59,250 @@ private List<IData> BuildDataList(Source[] paths, int batchSize)
+                                            correspondingTileBuilders.Add(
+                                                () => source.GetCorrespondingTile(coord, true));
+                                        }
 
-                                            // Create tile builder list for current coord for all sources
-                                            List<CorrespondingTileBuilder> correspondingTileBuilders =
-                                                new List<CorrespondingTileBuilder>();
-                                            foreach (IData source in sources)
-                                            {
-                                                correspondingTileBuilders.Add(
-                                                    () => source.GetCorrespondingTile(coord, true));
-                                            }
+                                        byte[]? blob = this._tileMerger.MergeTiles(correspondingTileBuilders, coord);
 
-                                            byte[]? blob = this._tileMerger.MergeTiles(correspondingTileBuilders, coord);
+                                        if (blob != null)
+                                        {
+                                            tiles.Add(new Tile(coord, blob));
+                                        }
 
-                                            if (blob != null)
-                                            {
-                                                tiles.Add(new Tile(coord, blob));
-                                            }
+                                        tileProgressCount++;
 
-                                            tileProgressCount++;
+                                        // Show progress every 1000 tiles
+                                        if (tileProgressCount % 1000 == 0)
+                                        {
+                                            this._logger.LogInformation($"Tile Count: {tileProgressCount} / {totalTileCount}");
 
-                                            // Show progress every 1000 tiles
-                                            if (tileProgressCount % 1000 == 0)
+                                            try
                                             {
-                                                this._logger.LogInformation($"Tile Count: {tileProgressCount} / {totalTileCount}");
+                                                task.Percentage = 100 * (tileProgressCount / totalTileCount);
+                                                taskUtils.UpdateProgress(task.JobId, task.Id, task.Percentage);
+                                            }
+                                            catch (Exception e)
+                                            {
+                                                this._logger.LogError($"Error in MergerService run - update task: {e.Message}");
```

Comment:
- update task: -> change to - update task percentage:

Reply:
Changed.